### PR TITLE
VESAlog and virtio-gpu improvements

### PR DESCRIPTION
**Pull request by StaticSaga**

### CODE REVIEW

File: src/platform/pci.zig [hunk 1]
```diff
@@ -145,8 +144,13 @@ fn function_scan(addr: Addr) void {
             os.log("Virtio display controller: Init has failed!\n", .{});
             return;
           };
+        if (os.drivers.vesa_log.get_info()) |vesa| {
           drv.modeset(os.drivers.vesa_log.framebuffer.?.bb_phys, vesa.width, vesa.height);
           os.drivers.vesa_log.set_updater(virtio_gpu.updater, @ptrToInt(drv));
+          os.log("Virtio display controller: Initialized with preexisting fb\n", .{});
+        } else {
+          os.drivers.vesa_log.register_fb(virtio_gpu.updater, @ptrToInt(drv), 800*4, 800, 600, 32);
+          drv.modeset(os.drivers.vesa_log.get_backbuffer_phy(), 800, 600);
```
Comment (N00byEdge) [inline src/platform/pci.zig:153]: Have you tested this without a framebuffer tag? Pretty sure the `framebuffer.?` inside vesa_log will fail.

Comment (StaticSaga) [reply src/platform/pci.zig:153]: The framebuffer is registered just before in `register_fb`

Comment (N00byEdge) [reply src/platform/pci.zig:153]: Sure, what I'm saying is that we never call `register_fb()` if we don't have a framebuffer tag from the bootloader.

**Review by N00byEdge — APPROVED** [review]
Works on your machine? ;)

**Comment by StaticSaga** [comment]
Haven't tested this one on bare metal, but I haven't touched anything that could break it.
It makes vesalog much faster when using TCG, which is pretty great especially with ARM (and maybe RISC-V too in the future!).
The main overhead now seems to be the virtio waits, required by the current setup with stack allocation; I will deal with it later, but performance is good enough for now.

**Pull request merged**